src/chm_lib.c

```diff
@@ -180,7 +184,7 @@ typedef unsigned long           UInt64;
 #endif
 
 /* GCC */
-#ifdef __GNUC__
+#if defined(__GNUC__) && !defined(WIN32)
```

Comment:
chm_lib.h and chm_lib.c are taken from the CHMLib project: https://github.com/jedwing/CHMLib
If indeed there are issues compiling the project on Windows, they should be fixed with a pull request in that project, and when Jed releases the next version of CHMLib (with your changes approved) we should take the updated files directly from his project.

Otherwise we would need to keep modifying those files manually for each new CHMLib release, in the xCHM project, and that's not ideal.

Reply:
CHMLib is not maintained, the last change was in 2009 and there are 6 open pull requests, but if you insist I can remove these changes and leave them in my fork.

Reply:
That's unfortunate, if that is indeed the case. In that case, what exactly is your commit fixing on Windows? If it's just compile-time warnings, I suggest we leave those alone and not bother with changing the CHMLib code.

FWIW, I did do xCHM Windows builds a long time ago with no issues (using Visual Studio).

Reply:
I use the mingw-w64 version of GCC from MSYS2 for building. This is the error message:

```
C:\src\xCHM\build-gcc>ninja
[2/16] Building C object CMakeFiles/xchm.dir/src/chm_lib.c.obj
C:/src/xCHM/src/chm_lib.c:76: warning: "strcasecmp" redefined
   76 | #define strcasecmp stricmp
      |
In file included from C:/src/xCHM/src/chm_lib.c:60:
C:/msys64/mingw64/include/string.h:120: note: this is the location of the previous definition
  120 | #define strcasecmp _stricmp
      |
C:/src/xCHM/src/chm_lib.c:77: warning: "strncasecmp" redefined
   77 | #define strncasecmp strnicmp
      |
C:/msys64/mingw64/include/string.h:119: note: this is the location of the previous definition
  119 | #define strncasecmp _strnicmp
      |
C:/src/xCHM/src/chm_lib.c: In function '_chm_decompress_region':
C:/src/xCHM/src/chm_lib.c:1545:27: warning: implicit declaration of function 'ffs' [-Wimplicit-function-declaration]
 1545 |         int window_size = ffs(h->window_size) - 1;
      |                           ^~~
C:/src/xCHM/src/chm_lib.c:1545:32: warning: 'ffs' argument 1 promotes to 'UInt32' {aka 'unsigned int'} where 'int' is expected in a call to built-in function declared without prototype [-Wbuiltin-declaration-mismatch]
 1545 |         int window_size = ffs(h->window_size) - 1;
      |                               ~^~~~~~~~~~~~~
<built-in>: note: built-in 'ffs' declared here
[16/16] Linking CXX executable xchm.exe
FAILED: xchm.exe
cmd.exe /C "cd . && C:\msys64\mingw64\bin\c++.exe -DCHM_MT -g -mwindows CMakeFiles/xchm.dir/src/chmapp.cpp.obj CMakeFiles/xchm.dir/src/chmfile.cpp.obj CMakeFiles/xchm.dir/src/chmfinddialog.cpp.obj CMakeFiles/xchm.dir/src/chmfontdialog.cpp.obj CMakeFiles/xchm.dir/src/chmframe.cpp.obj CMakeFiles/xchm.dir/src/chmfshandler.cpp.obj CMakeFiles/xchm.dir/src/chmhtmlnotebook.cpp.obj CMakeFiles/xchm.dir/src/chmhtmlwindow.cpp.obj CMakeFiles/xchm.dir/src/chmindexpanel.cpp.obj CMakeFiles/xchm.dir/src/chminputstream.cpp.obj CMakeFiles/xchm.dir/src/chmlistctrl.cpp.obj CMakeFiles/xchm.dir/src/chmsearchpanel.cpp.obj CMakeFiles/xchm.dir/src/hhcparser.cpp.obj CMakeFiles/xchm.dir/src/chm_lib.c.obj CMakeFiles/xchm.dir/src/lzx.c.obj CMakeFiles/xchm.dir/rc/xchm.rc.obj -o xchm.exe -Wl,--out-implib,libxchm.dll.a -Wl,--major-image-version,0,--minor-image-version,0  -LC:/msys64/mingw64/lib  -pipe  -Wl,--subsystem,windows  -mwindows  -lwx_mswu_core-3.2  -lwx_baseu-3.2  -lwx_mswu_aui-3.2  -lwx_mswu_html-3.2  -lwx_baseu_net-3.2  -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 && cd ."
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/13.1.0/../../../../x86_64-w64-mingw32/bin/ld.exe: CMakeFiles/xchm.dir/src/chm_lib.c.obj: in function `_chm_decompress_region':
C:/src/xCHM/src/chm_lib.c:1545: undefined reference to `ffs'
collect2.exe: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

Reply:
But MinGW already comes with CHMLib pre-packaged, so in this case I think the solution would be to use that external library rather than compiling the source code that comes with xCHM. You can use the `--enable-builtin-chmlib` configure flag for this.

https://packages.msys2.org/package/mingw-w64-ucrt-x86_64-chmlib?repo=ucrt64

I don't think merging changes to the local CHMLib source code is a good idea.

Reply:
I removed it from the PR.